lib/msal-browser/docs/working-with-b2c.md

```diff
@@ -0,0 +1,104 @@
+- Users **cannot** obtain tokens for Microsoft APIs (e.g. MS Graph API) using [delegated permissions](#b2c-and-delegated-permissions).
+- Applications **can** obtain tokens for Microsoft APIs using [application permissions](#b2c-and-application-permissions) (management scenarios).
+
+## B2C App Configuration
+
+The following is a B2C app configuration example:
+
+```javascript
+const msalConfig = {
+  auth: {
+    clientId: "<your-clientID>",
+    authority: "https://<your-tenant>.b2clogin.com/<your-tenant>.onmicrosoft.com/<your-policyID>",
+    knownAuthorities: ["<your-tenant>.b2clogin.com"] // array of URIs that are known to be valid
+  }
+}
+
+const apiConfig = {
+  b2cScopes: ["https://<your-tenant>.onmicrosoft.com/<your-api>/<your-scope>"],
+  webApiUri: "<your-api-uri>" // e.g. "https://fabrikamb2chello.azurewebsites.net/hello"
+};
+
+const loginRequest = {
+  scopes: [ "openid", "offline_access" ]
+}
+
+const tokenRequest = {
+  scopes: apiConfig.b2cScopes // e.g. "https://<your-tenant>.onmicrosoft.com/<your-api>/<your-scope>"
+}
+```
+
+## AAD vs B2C Endpoints
+
+A major difference between **Azure AD** (*AAD* for short) vs. **B2C** tenants are with respect to their **endpoints**.
+
+An **AAD** tenant:
+
+- Contains only AAD endpoints (`login.microsoftonline.com/*`).
+- Exposes a **single** token endpoint (`login.microsoftonline.com/.../token`).
+- AAD endpoints allow you to obtain tokens for:
+  - Your applications protected by AAD.
+  - Microsoft APIs, such as **MS Graph API**.
+
+A **B2C** tenant:
+
+- Contains AAD and B2C endpoints (`login.microsoftonline.com/*` and `<your-domain>.b2clogin.com/*`).
+- Exposes **separate** token endpoints for each (`login.microsoftonline.com/.../token`, `<your-domain>.b2clogin.com/.../token`).
+- B2C endpoints allow you to obtain tokens for:
+  - Your applications protected by B2C.
+
+## B2C and Delegated Permissions
+
+**Delegated permissions** specify *scope-based access* using *interactive* authorization from the signed-in user. These permissions are presented to the resource (*e.g.* your web API, MS Graph API etc.) at run-time as `scp` claims in the client's **Access Token**.
+
+A user's B2C authentication cannot be used to authorize to AAD protected apps, or **Microsoft APIs** (which are also protected by AAD). As such, when you use **MSAL.js**, you cannot use the `<your-tenant>.b2clogin.com/.../token` endpoint to obtain a token for **MS Graph API**.
+
+### OpenID Connect Permissions
+
+The exception to the rule above comes from a special set of scopes known as **OpenID Connect** (OIDC) permissions, which includes `openid` and `profile`. Another special permission is the `offline_access`, which gives your app access to a resources on behalf of the user for an extended time (using a **Refresh Token**). MSAL.js will supply `openid`, and `profile` by default during `loginPopup()` and `loginRedirect()` requests.
```

Comment:
Why is this an exception? It is confusing to say OIDC permissions are exception, as they mean different things for B2C endpoint and MS Graph endpoint. Should we rephrase this?

Reply:
Hmm, by exception I meant that even though they are listed under MS Graph permissions on the Azure Portal, they're not used for any Graph resource (/userinfo is technically hosted on Graph but...). 

I'm thinking that until the Portal UI updates, people will be confused why these scopes and not other Graph scopes. I think once that's done we can remove this explanation.. -or do you have a suggestion for rephrase?